xbmc/filesystem/FileCache.cpp

```diff
@@ -277,13 +277,13 @@ void CFileCache::Process()
 
     while (m_writeRate)
     {
-      if (m_writePos - m_readPos < m_writeRate)
+      if (m_writePos - m_readPos < m_writeRate * g_advancedSettings.m_readBufferFactor)
```

Comment:
Careful here. Think of a raw bluray a user is playing and having a readBufferFactor of 8 over 100 Mbit/s Lan - that would constantly reset the limiter.